GetStream/stream-chat-android
# [3793] Refactor QueryChannelsPostponeHelper to make it event driven

### 🎯 Goal

Make `QueryChannelsPostponeHelper` event-driven.

### 🧪 Testing

- Run any sample app
- Login
- Wait until ChatList gets shown
- Kill the app
- Filter Logcat by `queryChannels`
- Reopen it again

Make sure you don't see any errors in Logcat


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
